microhapulator/mapstats.py

```diff
@@ -0,0 +1,100 @@
+# -------------------------------------------------------------------------------------------------
+# Copyright (c) 2024, DHS.
+#
+# This file is part of MicroHapulator (https://github.com/bioforensics/microhapulator) and is
+# licensed under the BSD license: see LICENSE.txt.
+#
+# This software was prepared for the Department of Homeland Security (DHS) by the Battelle National
+# Biodefense Institute, LLC (BNBI) as part of contract HSHQDC-15-C-00064 to manage and operate the
+# National Biodefense Analysis and Countermeasures Center (NBACC), a Federally Funded Research and
+# Development Center.
+# -------------------------------------------------------------------------------------------------
+
+from collections import namedtuple, defaultdict
+from dataclasses import dataclass
+import pandas as pd
+from pathlib import Path
+import re
+from typing import Dict
+
+
+class MappingSummary(dict):
+    @classmethod
+    def from_workdir(cls, samples, workdir="."):
+        summary = cls()
+        for sample in sorted(samples):
+            summary[sample] = MappingStats.from_workdir(sample, workdir=workdir)
+        return summary
+
+    @property
+    def markers(self):
+        for stats in self.values():
+            return sorted(stats.repetitive.keys())
+
+    def repetitive_reads_by_marker(self):
+        counts = defaultdict(dict)
+        for marker in self.markers:
+            for sample, stats in self.items():
+                counts[marker][sample] = stats.read_count_pair(marker)
+        return counts
```

Comment:
Nested dictionary data structure for populating the table of repetitive reads per marker per sample.